ank/src/cli_commands/get_configs.rs

```diff
@@ -0,0 +1,159 @@
+// Copyright (c) 2024 Elektrobit Automotive GmbH
+//
+// This program and the accompanying materials are made available under the
+// terms of the Apache License, Version 2.0 which is available at
+// https://www.apache.org/licenses/LICENSE-2.0.
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations
+// under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+use super::CliCommands;
+use crate::cli_commands::config_table_row::ConfigTableRow;
+use crate::filtered_complete_state::FilteredCompleteState;
+use crate::{cli_commands::cli_table::CliTable, cli_error::CliError, output_debug};
+use common::objects::ConfigItem;
+const EMPTY_FILTER_MASK: [String; 0] = [];
+
+impl CliCommands {
+    // [impl->swdd~cli-provides-list-of-configs~1]
+    // [impl->swdd~cli-processes-complete-state-to-provide-connected-agents~1]
+    pub async fn get_configs(&mut self) -> Result<String, CliError> {
+        let filtered_complete_state: FilteredCompleteState = self
+            .server_connection
+            .get_complete_state(&EMPTY_FILTER_MASK)
+            .await?;
```

Comment:
We can actually make the request here with a filter mask specifying only the configs: `"desiredState.configs`.
It's not such a big improvement, but if the complete state is large, we will spare some unnecessary data transfer.

Please don't forget to also update the diagram above accordingly.

Reply:
Now the filter described by the `DESIRED_STATE_CONFIGS` is being used.